game-app/game-core/src/main/java/games/strategy/engine/data/util/BreadthFirstSearch.java

```diff
@@ -27,19 +27,10 @@ public abstract static class Visitor {
      * Called when a new territory is encountered.
      *
      * @param territory The new territory.
-     */
-    public abstract void visit(Territory territory);
-
-    /**
-     * Called after all territories within the specified distance have been searched. Can be
-     * overridden to terminate the search.
-     *
-     * @param distanceSearched The current distance searched
+     * @param distance The distance to the territory.
      * @return Whether the search should continue.
      */
-    public boolean shouldContinueSearch(final int distanceSearched) {
-      return true;
-    }
+    public abstract boolean visit(Territory territory, int distance);
```

Comment:
Without a default implementation this can be turned into an interface. If you prefer you can also annotate it with `@FunctionalInterface` to ensure the interface can be turned into a lambda

Reply:
Good point, done!